src/core/worker.js

```diff
@@ -623,6 +508,16 @@ var WorkerMessageHandler = {
             }
 
             if (pdfManager) {
+              // There may be a chance that pdfManager is not initialized
+              // for some previous runs of this(readchunk) block of code.
+              // Be sure to send all cached chunks, if any, to chunked_stream
+              // via pdf_manager.
+              if (cachedChunks.length !== 0) {
+                for (let i = 0; i < cachedChunks.length; i++) {
+                  pdfManager.sendProgressiveData(cachedChunks[i]);
+                }
+                cachedChunks = [];
+              }
```

Comment:
Done. Thanks. Also I think we don't need to check for `if (cachedChunks.length !== 0) {..`, as it will not invoke `pdfManager.sendProgressiveData` if `cachedChunks.length === 0`